### DISCUSSION TITLE
【V5】引入G6-5.x报错TypeError: Cannot read properties of undefined (reading 'Graph')

### DISCUSSION THREAD

**Discussion by TZZack** [antvis/G6#6504] [Q&A] [answered]
G6版本：5.0.29
看到issue那边也有人提了个，但是后面closed了，没有有效信息
https://github.com/antvis/G6/issues/6049
代码是很简单的一个demo，如下：
```
<template>
  <div id="containerId1" class="research-graph"></div>
</template>

<script lang="ts" setup>
import { Graph } from '@antv/g6';

const data = {
  nodes: [
    {
      id: 'node-1',
      style: { x: 50, y: 100 },
    },
    {
      id: 'node-2',
      style: { x: 150, y: 100 },
    },
  ],
  edges: [{ id: 'edge-1', source: 'node-1', target: 'node-2' }],
};

const graph = new Graph({
  container: 'containerId1',
  data,
});

graph.render();

</script>
```

**Comment by Aarebecca** [comment]
是无法启动还是构建时出错

**Reply by TZZack** [reply, marked as answer]
感谢回复，非G6问题，我忘记回来补充了，是我这边的问题，排查过程和结果如下：
1、最终找到控制台实际报错的地方是`dagre`这个依赖里面的`var Graph = require("./graphlib").Graph`这行代码报的错，也就是require了个undefined进来
2、graphlib.js文件里面就是把`graphlib`这个库引入进来而已（这里有try catch），然后就接着往`graphlib`这个库排查
3、最终发现是`graphlib`里面的`dijkstra.js`这个文件的`var DEFAULT_WEIGHT_FUNC = _.constant(1)`报错了，因为这里的lodash函数编译处理后多了一层default，所以`_.constant.default(1)`才会正常
4、那为什么多了一层default呢，看着是lodash有问题就想起这个项目有个`webpack插件`把所有的lodash的import路径都改为`lodash-es`，graphlib里面的lodash是这样引入的`var lodash = { ..., constant: require("lodash/constant") }`，也就是变为了lodash-es/constant，然后graphlib是cjs，就可能是webpack处理的时候没处理好吧（这个目前没时间深究，后续有时间继续排查下）
（目前是把这个插件改了下，不处理cjs库的lodash了）

**Reply by ifyour** [reply]
感谢，问题解决，把项目内的 lodash 按正常引入即可，别使用 lodash -> lodash-es 的这层映射。
